File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    int len = nums.size();
    vector<bool> check(200000);
    
    for(int i = 0; i < len ; i++){
        if(answer >= len/2) break;
        if(check[nums[i]] == false){
            check[nums[i]] = true;
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = nums.length/2;
        HashSet<Integer> numSet = new HashSet<>();
        for(int i : nums) numSet.add(i);
        answer = Math.min(numSet.size(), answer);
        return answer;
    }
}